common/jsonquery/defaults.go

```diff
@@ -26,6 +31,31 @@ func (q *Query) StrWithDefault(key string, defaultValue string) (string, error)
 	return *result, nil
 }
 
+// TextWithDefault returns a string under the key regardless of JSON data type.
+// If data is not a string it will be converted to such.
+func (q *Query) TextWithDefault(key string, defaultValue string) (string, error) {
```

Comment:
When extracting "id" from JSON we don't care if it is string or integer, eventually we represent it as a string. This JSON data type distinction is irrelevant for our goals. If it matters other methods should be used Integer/Str/Bool.